pkg/cluster/k8sres.go

```diff
@@ -669,6 +669,48 @@ func (c *Cluster) generateSingleUserSecret(namespace string, pgUser spec.PgUser)
 	return &secret
 }
 
+func (c *Cluster) shouldCreateLoadBalancerForService(role PostgresRole, spec *spec.PostgresSpec) bool {
+
+	switch role {
+
+	case Replica:
+
+		// deprecated option takes priority for backward compatibility
+		if spec.ReplicaLoadBalancer != nil {
+			c.logger.Debugf("The Postgres manifest for the cluster %v sets the deprecated `replicaLoadBalancer` param. Consider using the `enableReplicaLoadBalancer` instead.", c.Name)
+			return *spec.ReplicaLoadBalancer
+		}
+
+		// if the value is explicitly set in a Postgresql manifest, follow this setting
+		if spec.EnableReplicaLoadBalancer != nil {
+			return *spec.EnableReplicaLoadBalancer
+		}
+
+		// otherwise, follow the operator configuration
+		return c.OpConfig.EnableReplicaLoadBalancer
+
+	case Master:
+
+		if spec.UseLoadBalancer != nil {
+			c.logger.Debugf("The Postgres manifest for the cluster %v sets the deprecated `useLoadBalancer` param. Consider using the `enableMasterLoadBalancer` instead.", c.Name)
+			return *spec.UseLoadBalancer
+		}
+
+		// `enable_load_balancer`` governs LB for a master service
+		// there is no equivalent operator configmap option for the replica LB
```

Comment:
`there is no equivalent...` this is confusing, taking into account `EnableReplicaLoadBalancer`, did you mean there is nothing like that in the old version?

Reply:
> did you mean there is nothing like that in the old version?

Yes, exactly. What I mean here is that in the deprecated set of params for configuring LB at the *operator* level - that is, for all Postgres clusters  the operator watches - there is no separate param for replica LBs.  Good catch :+1: